tests/e2e/test_accessibility.py

```diff
@@ -0,0 +1,128 @@
+    ]
+
+    def test_w3_validator(self):
+        # Uncomment this to generate a new baseline for all pages on the website
+        # Then run 'python3 -m unittest e2e.test_accessibility' from inside the tests folder
+        # genBaseline(self)
+
+        # Uncomment this to generate a new baseline for a specific url
+        # url = '' # your url here
+        # genBaseline(self, url)
+
+
+        validatePages(self)
+
+
+
+def validatePages(self):
+    self.log_out()
+    self.log_in(user_id='instructor')
+    self.click_class('sample')
+    with open('/'.join(__file__.split('/')[:-1])+'/accessibility_baseline.json') as f:
+        baseline = json.load(f)
+
+    foundError = False
+    for url in self.urls:
+        self.get(url=url)
+
+        payload = self.driver.page_source
+        headers = {
+          'Content-Type': 'text/html; charset=utf-8'
+        }
+        response = requests.request("POST", "https://validator.w3.org/nu/?out=json", headers=headers, data = payload.encode('utf-8'))
+
+
+
+        for error in response.json()['messages']:
+            # For some reason the test fails to detect this even though when you actually look at the rendered
+            # pages this error is not there. So therefore the test is set to just ignore this error.
+            if error['message'].startswith("Start tag seen without seeing a doctype first"):
+                continue
+
+            if error['message'] not in baseline[url]:
+                error['url'] = url
+                print(json.dumps(error, indent=4, sort_keys=True))
+                foundError = True
+
+    self.assertEqual(foundError, False)
+
+
+
+def genBaseline(self, new_url=None):
+    self.log_out()
+    self.log_in(user_id='instructor')
+    self.click_class('sample')
+
+    baseline = {}
+    urls = self.urls
+    if new_url:
+        with open('/'.join(__file__.split('/')[:-1])+'/accessibility_baseline.json') as f:
+            baseline = json.load(f)
+        urls = [new_url]
+
+    for url in urls:
+        self.get(url=url)
+        payload = self.driver.page_source
+        headers = {
+          'Content-Type': 'text/html; charset=utf-8'
+        }
+        response = requests.request("POST", "https://validator.w3.org/nu/?out=json", headers=headers, data = payload.encode('utf-8'))
+
+        if new_url == None or url == new_url:
+            baseline[url] = {}
+        for error in response.json()['messages']:
+            # For some reason the test fails to detect this even though when you actually look at the rendered
+            # pages this error is not there. So therefore the test is set to just ignore this error.
+            if error['message'].startswith("Start tag seen without seeing a doctype first"):
+                continue
+
+            if error['message'] not in baseline[url]:
+                baseline[url][error['message']] = error
+    with open('/'.join(__file__.split('/')[:-1])+'/accessibility_baseline.json', 'w') as file:
```

Comment:
Should we be using `os.path.dirname(os.path.realpath(__file__))` instead? This will always ensure the full path is used rather than the relative.